drift/info_parser.py

```diff
@@ -129,6 +129,21 @@ def build_comparisons(
     }
 
 
+def _set_summary_state(grouped_comparison):
```

Comment:
Can you please also refactor it to take the list of states and return a computed state? It would read much better and would not affect the input data structure.

Reply:
That's what this code does. :)
```
states = {comparison["state"] for comparison in grouped_comparison["comparisons"]}
```

Here return a list of state in that group, and return computed state after.

Reply:
This function does a side effect on the input. My suggestion is to do a return value and no side effect.

State gathering should move out
```
states = {comparison["state"] for comparison in grouped_comparison["comparisons"]}
```

and the function should look like this:
```
def summary_state(states):
    if COMPARISON_DIFFERENT in states:
        return COMPARISON_DIFFERENT
    elif COMPARISON_INCOMPLETE_DATA in states:
    ...
``` 

then outside of the function
```
group_states = {comparison["state"] for comparison in group["comparisons"]}
group["state"] = summary_state(group_states)
```

Reply:
Make sense, fixed on this commit: c46c812f9e26e5d42f939b7e5b8dd7b256b365d7